### PR TITLE
Tilt

### PR DESCRIPTION
**Pull request by guilbill**
Tilt board should move pucks. See [card](https://trello.com/c/T2HtA2z6/1-1-as-pierre-i-would-like-to-be-able-to-play-basic-tipsy-games-in-my-browser)
tipsy-basic

### CODE REVIEW

**Comment by fzaninotto** [comment]
I would love a screencast!

File: src/Controller/GameController.php [hunk 1]
```diff
@@ -5,20 +5,58 @@
...
+    {
+        $playerHash = $request->cookies->get($this::COOKIE_KEY);
+        $board = $this->session->get($playerHash);
         return $this->render('game/game.html.twig', [
-            'board' => $this->gameService->newGame(),
+            'board' => $board
         ]);
     }
+
+    public function action(Request $request)
```
Comment (fzaninotto) [inline src/Controller/GameController.php]: not fan of this name, too generic

File: src/Entity/Board.php [hunk 2]
```diff
@@ -87,15 +89,92 @@ public function getPuck($x, $y)
...
+        $puckId = array_search($puck, $this->graph->vertices);
+        if (!empty($this->graph->vertices[$puckId])) {
+            $this->graph->vertices[$puckId]->setValue();
+        }
+        return [$color];
+    }
+
+    public function movePuckTo($puck, $direction)
+    {
+        // neighbor = self.__get_node_by_direction(node, direction)
```
Comment (fzaninotto) [inline src/Entity/Board.php]: commented out code

**Review by fzaninotto — CHANGES_REQUESTED** [review]

File: src/Controller/GameController.php [hunk 3]
```diff
@@ -4,21 +4,63 @@
...
+        $response = $this->redirectToRoute('game');
+        $response->headers->setCookie(new Cookie($this::COOKIE_KEY, $playerHash));
+        $this->session->set($playerHash, $game);
+
+        return $response;
+    }
+
+    public function show(Request $request)
+    {
+        $playerHash = $request->cookies->get($this::COOKIE_KEY);
```
Comment (alexisjanvier) [inline src/Controller/GameController.php:39]: What happens if there is no cookie?

File: src/Controller/GameController.php (hunk 1 above)
Comment (guilbill) [reply src/Controller/GameController.php]: Agree, changed it to `tilt`

**Review by alexisjanvier — CHANGES_REQUESTED** [review]
the absence of a cookie causes an error on the route game

File: src/Controller/GameController.php (hunk 3 above)
Comment (guilbill) [reply src/Controller/GameController.php:39]: I now right? :P
Do you think I should redirect to `/` ? In order to create a new game... That's what seems best to me...

File: src/Controller/GameController.php [hunk 4]
```diff
@@ -37,6 +37,9 @@ public function new()
     public function show(Request $request)
     {
         $playerHash = $request->cookies->get($this::COOKIE_KEY);
+        if(!$playerHash){
```
Comment (alexisjanvier) [inline src/Controller/GameController.php:40]: missing test, but ok

**Review by alexisjanvier — APPROVED** [review]

**Pull request merged**